spec/requests/document_sets_controller_spec.rb

```diff
@@ -0,0 +1,151 @@
+    end
+  end
+
+  describe '#settings_works' do
+    let(:action_path) { collection_settings_works_path(owner, document_set) }
+
+    subject { get action_path }
+
+    it 'renders status and template' do
+      login_as owner
+      subject
+
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:settings_works)
+    end
+  end
+
+  describe '#update' do
+    let(:params) { {} }
+    let(:action_path) { document_set_path(document_set, collection_id: document_set.slug) }
+    let(:xhr) { false }
+
+    subject { put action_path, params: params, xhr: xhr }
+
+    context 'image upload' do
+      let(:params) do
+        {
+          document_set: {
+            picture: Rack::Test::UploadedFile.new(Rails.root.join('test_data/images/pages/sanskrit.jpg'), 'images/jpg')
+          }
+        }
+      end
+
+      it 'renders status and template' do
+        login_as owner
+        subject
+
+        follow_redirect!
+        expect(response).to have_http_status(:ok)
+        expect(response).to render_template(:settings)
+      end
+    end
+
+    context 'valid params' do
+      let(:params) do
+        {
+          document_set: {
+            title: 'New title',
+            description: 'New description'
+          }
+        }
+      end
+      let(:xhr) { true }
+
+      it 'renders status and json response' do
+        login_as owner
+        subject
+        expect(response).to have_http_status(:ok)
+        expect(JSON.parse(response.body, symbolize_names: true)).to include(
+          success: true,
+          errors: nil,
+          updated_field: {
+            title: 'New title',
+            description: 'New description',
+            slug: 'new-title'
+          }
+        )
+      end
+    end
+
+    context 'invalid params' do
+      let(:params) do
+        {
+          document_set: {
+            title: ''
+          }
+        }
+      end
+
+      it 'renders status and template' do
+        login_as owner
+        subject
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response).to render_template(:settings)
+      end
+    end
+  end
+
+  describe '#toggle_privacy' do
+    let(:action_path) { document_sets_toggle_privacy_path(collection_id: document_set) }
+
+    subject { post action_path }
+
+    it 'renders status and template' do
+      login_as owner
+      subject
+
+      follow_redirect!
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:settings_privacy)
+    end
+  end
+
+  describe '#destroy' do
+    let(:action_path) { document_sets_destroy_path(document_set_id: document_set.id) }
+
+    subject { delete action_path }
+
+    it 'renders status and template' do
+      login_as owner
+      subject
+
+      follow_redirect!
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:index)
+    end
+  end
+end
```

Comment:
Ideally, we add all defined routes/functions inside document_sets_controller here. That way we have a clear test on all requests.